### PR TITLE
GRIDEDIT-291: Make sure that we can compile MeshKernel on CentOS 7

### PR DESCRIPTION
**Pull request by lucacarniato**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,4 @@ xbuild
 
 # User specific
 post_build.bat
+/tests/utils/include/TestUtils/Definitions.hpp
```
Comment (Hofer-Julian) [inline .gitignore:24]: What is that one good for?

File: CMakeLists.txt [hunk 2]
```diff
@@ -6,8 +6,8 @@ project(
   MeshKernel
   VERSION 0.10.0
   DESCRIPTION "Library for creating and editing meshes."
-  LANGUAGES CXX)
-
+  LANGUAGES CXX C)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```
Comment (Hofer-Julian) [inline CMakeLists.txt:10]: Why is this necessary?

File: CMakeLists.txt [hunk 3]
```diff
@@ -17,7 +17,7 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
     set(CMAKE_CXX_FLAGS_RELEASE
         "-g -O2"
         CACHE STRING "List of C++ compiler flags for a Release build")
```
Comment (Hofer-Julian) [inline CMakeLists.txt:19]: Why don't you specify c++17 in the release build as well?

File: CMakeLists.txt [hunk 4]
```diff
@@ -44,9 +44,23 @@ add_subdirectory(extern)
...
+endif()  
+if(WIN32)
+  # In windows use static libraries
+  set(Boost_USE_STATIC_LIBS ON)
+  find_package(Boost REQUIRED)
+endif()  
 include_directories(${Boost_INCLUDE_DIR})
 
+#NetCDF
+find_path(NETCDF_BIN_PATH NAMES "${CMAKE_SHARED_LIBRARY_PREFIX}netcdf${CMAKE_SHARED_LIBRARY_SUFFIX}"
```
Comment (Hofer-Julian) [inline CMakeLists.txt:59]: We do this only on Windows, right?
So make add an if(Win32) to it

File: src/MeshKernelApi/MeshKernel.cpp [hunk 5]
```diff
@@ -123,8 +124,7 @@ namespace meshkernelapi
 
         for (const auto& index : indices)
         {
-            const auto size = index[1] - index[0] + 1;
-            if (size > 0)
+            if (const auto size = index[1] - index[0] + 1; size > 0)
```
Comment (Hofer-Julian) [inline src/MeshKernelApi/MeshKernel.cpp:127]: I still don't like this syntax...

**Review by Hofer-Julian — COMMENTED** [review]
Just a few small comments

File: CMakeLists.txt (hunk 2 above)
Comment (lucacarniato) [reply CMakeLists.txt:10]: bcs triangle is in C, and g++ gnu compiler seems not able to compile .c files. we need to explicitly state is also a c project

File: CMakeLists.txt (hunk 3 above)
Comment (lucacarniato) [reply CMakeLists.txt:19]: good point, i will add it

File: src/MeshKernelApi/MeshKernel.cpp (hunk 5 above)
Comment (lucacarniato) [reply src/MeshKernelApi/MeshKernel.cpp]: I see your point if I can't inspect the value of size when debugging. if this is the case we can revert this

File: CMakeLists.txt (hunk 2 above)
Comment (Hofer-Julian) [reply CMakeLists.txt:10]: I meant the "CMAKE_POSITION_INDEPENDENT_CODE ON" part.
Could you please elaborate on that?
